staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -4609,6 +4623,19 @@ func (c *Cloud) getFullInstance(nodeName types.NodeName) (*awsInstance, *ec2.Ins
 	return awsInstance, instance, err
 }
 
+func (c *Cloud) nodeNameToProviderID(nodeName types.NodeName) (InstanceID, error) {
+	// TODO: get node from cache
```

Comment:
This PR looks reasonably good as the first step, but we really need node cache here. Look at what GCE cloud provider does.

Reply:
Agree. I filed #78409 to track the progress